template-for-repository/proofs/arpa-test.sh

```diff
@@ -0,0 +1,342 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script performs  data collection for Arpa.
+# Furthermore, it serves as a validation framework for existing CBMC proofs.
+
+# TODO add project-specific list of proofs that fail, but should be overriden due to limitations of the validation approach
+# The list below must be seperated by spaces
+manual_proof_override=""
+
+
+function initialize {
+    # variable initializations
+    results=arpa-test-results.log
+    arpa_log=arpa-test-logs
+    t_pfs_w_unnesc=0
+    tot_prfs=$(ls -l | grep -c ^d)
+
+    # remove existing results
+    rm -f $results
+
+    # consider command-line flag
+    if [[ "$1" == "-mo" || "$1" == "--manual-override" ]]; then
+
+        echo -e "\n<MANUAL OVERRIDING ENABLED>"
+        manual_override_flag=1
+    fi
+}
+
+
+function look_for {
+    if [ ! -f $1 ]; then
+        echo "-- $1 does not exist in ($dir)." 1>&2
+        echo "< SKIPPING >" 1>&2
+        cd ..
+        echo "($cnt/$tot_prfs) - ERROR - $dir" >> $results
+        echo -e "  0.- $1 does not exist in ($dir)\n" >> $results
+        continue
+    fi
+}
+
+
+function proof_init {
+    # proof-specific initialisations
+    cd $dir
+    harness="$dir"_harness.c
+    rm -r -f $arpa_log
+    echo -e "\n<($cnt/$tot_prfs)-BEGIN-($dir)>"
+
+    # check if required files exist
+    look_for Makefile
+    look_for $harness
+}
+
+
+function get_fcts {
+    # this function gets the goto functions relevant to the CBMC proof
+    # and store it in a variable
+    goto_fcts=$(cbmc --show-goto-functions gotos/$harness.goto)
+
+    # get rid of comments
+    clean_goto_fcts=$(grep -v " *//.*" <<< "$goto_fcts")
+    eval fcts_$1_clean="\$clean_goto_fcts"
```

Comment:
You see to be invoking eval on unsanitized input $1.  That seems very dangerous.